### ISSUE TITLE
[studio] Content monitoring does not work

### ISSUE THREAD

**Issue by alhambrav** [craftercms/craftercms#1128]
I tried following the instructions in the doc to do content monitoring:
http://docs.craftercms.org/en/3.0/site-administrators/studio/content-monitoring.html

I edited the `content-monitoring.groovy`, by uncommenting the code there
 and I get an error when the file `content-monitoring.groovy` is run 
```
[INFO] 2017-07-20 14:50:00,003 [craftercms_scheduler_Worker-3] [core.JobRunShell] | Job DEFAULT.craftercms:/scripts/jobs/every-5-minutes/content-monitoring.groovy threw a JobExecutionException:
org.quartz.JobExecutionException: Error executing script job at /scripts/jobs/every-5-minutes/content-monitoring.groovy [See nested exception: org.craftercms.engine.exception.ScriptException: No such property: userName for class: org.craftercms.studio.impl.v1.service.deployment.job.DeployContentToEnvironmentStore]
	at org.craftercms.engine.scripting.impl.ScriptJob.execute(ScriptJob.java:47)
	at org.quartz.core.JobRunShell.run(JobRunShell.java:202)
	at org.quartz.simpl.SimpleThreadPool$WorkerThread.run(SimpleThreadPool.java:573)
Caused by: org.craftercms.engine.exception.ScriptException: No such property: userName for class: org.craftercms.studio.impl.v1.service.deployment.job.DeployContentToEnvironmentStore
	at org.craftercms.engine.scripting.impl.GroovyScript.execute(GroovyScript.java:61)
	at org.craftercms.engine.scripting.impl.ScriptJob.execute(ScriptJob.java:45)
	... 2 more
Caused by: groovy.lang.MissingPropertyException: No such property: userName for class: org.craftercms.studio.impl.v1.service.deployment.job.DeployContentToEnvironmentStore
	at org.codehaus.groovy.runtime.ScriptBytecodeAdapter.unwrap(ScriptBytecodeAdapter.java:53)
	at org.codehaus.groovy.runtime.callsite.GetEffectivePojoPropertySite.getProperty(GetEffectivePojoPropertySite.java:66)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callGetProperty(AbstractCallSite.java:296)
	at content-monitoring.run(content-monitoring.groovy:28)
	at groovy.util.GroovyScriptEngine.run(GroovyScriptEngine.java:605)
	at org.craftercms.engine.scripting.impl.GroovyScript.execute(GroovyScript.java:55)
	... 3 more
```

**Comment by cortiz** [comment]
@sumerjabri and or @dejan-brkic The issue here is bigger, the Notification Workflow config file  is only loaded when site is created, if the tomcat is reloaded this config is never reloaded and it will stop to work, this also includes notification emails and anything else related to that file `notification-config.xml` @dejan-brkic any pointers on where the configuration is load when a site is already created?

**Comment by jdrossl** [comment]
@sumerjabri @dejan-brkic I opened a new ticket for the issue with notifications because it affects all notifications not just the content monitors.

**Comment by alhambrav** [comment]
@jdrossl 
Hi! I asked Sumer about what we should do about the notifications sent out and how often should the process run.  Here's a summary:

Allow multiple queries when configuring content monitoring.  Also, the queries should not be a range.  We can have queries like the following:
- warn 90 days before the content will expire
- warn 60 days before the content will expire
- warn 30 days before the content will expire
- warn 7 days before the content will expire
- warn 1 day before the content will expire

The content monitoring process should run only once a day. So, if a query has been met, the notification email is only sent once.  And since we can have multiple queries, other notification emails can be sent out depending on what time  before expiry was specified in the query.  We can then also just have multiple notification email templates for each query listed in the content monitoring configuration.

With the above in mind, say we have an article named `a1` that will expire in time `t1`, let's set `t1 = 0` the time the article `a1` expires.  Let's say, we have another article `a2`, that expires 6 days after `a1` expires, so we'll define `t2 = t1 + 6`

If we have the queries like listed above in the configuration, at `t1 -90`, the first query to warn 90 days before the content will expire, will send out a notification email that article `a1` is about to expire in 90 days.  

At `t1 - 60`, the query to warn 60 days before the content will expire, will send out a notification email that article `a1` will expire in 60 days, and so on.  

When the time is `t1 - 1`, the query to warn 1 day before the content will expire, will send out a notification email that article `a1` will expire in 1 day and at the same time, the query to warn 7 days before the content will expire, will send out a notification email that article `a2` will expire in 7 days.

If you have any questions just let me or @sumerjabri know.

**Comment by jdrossl** [comment]
@alhambrav @sumerjabri in the current implementation is possible to have multiple queries, however all monitors will run at the same time (once a day according to the change) and the queries are completely open to the site admins (they do not have to be ranges, that was only an example).
Is that ok or do we need to make a bigger change in how the queries are defined?

**Comment by alhambrav** [comment]
Verified.
